processfamily/__init__.py

```diff
@@ -246,68 +254,45 @@ def __init__(self, process_instance, echo_std_err, child_index, process_family):
+        if self.MONITOR_STDOUT:
+            self._sys_out_thread = threading.Thread(target=self._sys_out_thread_target, name="pf_%s_stdout" % self.name)
+            self._sys_out_thread.daemon = True
             self._sys_out_thread.start()
 
-    def send_command(self, command, timeout, params=None):
-        response_id = str(uuid.uuid4())
-        try:
-            self._send_command_req(response_id, command, params=params)
-            return self._wait_for_response(response_id, timeout)
-        finally:
-            self._cleanup_queue(response_id)
-
-    def _send_command_req(self, response_id, command, params=None):
-        with self._rsp_queues_lock:
-            if self._rsp_queues is None:
-                return
-            self._rsp_queues[response_id] = Queue.Queue()
-        cmd = {
-            "method": command,
-            "id": response_id,
-            "jsonrpc": "2.0"
-        }
-        if params is not None:
-            cmd["params"] = params
-
-        req = json.dumps(cmd)
-        if '\n' in req:
-            raise ValueError('Invalid request string (new lines are not allowed): "%r"' % req)
-
-        try:
-            with self._stdin_lock:
-                self._process_instance.stdin.write("%s\n" % req)
-                self._process_instance.stdin.flush()
-                if command == 'stop':
-                    #Now close the stream - we are done
-                    self._process_instance.stdin.close()
-        except Exception as e:
-            if self._process_instance.poll() is None:
-                #The process is running, so something is wrong:
-                raise
-
-    def _wait_for_response(self, response_id, timeout):
-        with self._rsp_queues_lock:
-            if self._rsp_queues is None:
-                return None
-            q = self._rsp_queues.get(response_id, None)
-        if q is None:
-            return None
-        try:
-            if timeout <= 0:
-                return q.get_nowait()
-            else:
-                return q.get(True, timeout)
-        except Queue.Empty as e:
-            return None
-
-    def _cleanup_queue(self, response_id):
-        with self._rsp_queues_lock:
-            if self._rsp_queues is not None:
-                self._rsp_queues.pop(response_id, None)
+    def __repr__(self):
+        return "%s (%s: %r)" % (self.name, type(self).__name__, self._process_instance)
+
+    @property
+    def pid(self):
+        return self._process_instance.pid
+
+    def is_stopped(self):
+        """return whether the governed process has stopped"""
+        return self._process_instance.poll() is not None
+
+    @staticmethod
+    def get_popen_streams(echo_std_err):
+        """Returns kwargs for stdin, stdout and stderr to pass to subprocess.Popen"""
+        PIPE = subprocess.PIPE
+        streams = {"stdin": PIPE, "stdout": PIPE, "stderr": PIPE}
+        if echo_std_err:
```

Comment:
Do we do the echoing elsewhere?  'If echo_stderr then redirect it to /dev/null' is confusing - if this is right, perhaps a comment is needed to explain it?